apps/bttester/src/btp_pacs.c

```diff
@@ -0,0 +1,300 @@
+
+    switch (char_cmd->char_id) {
+        case BTP_PACS_CHARACTERISTIC_SINK_PAC:
+            rc = ble_audio_codec_register(&snk_codec_params, NULL);
+            if (rc) {
+                return BTP_STATUS_FAILED;
+            }
+        case BTP_PACS_CHARACTERISTIC_SOURCE_PAC:
+            rc = ble_audio_codec_register(&src_codec_params, NULL);
+            if (rc) {
+                return BTP_STATUS_FAILED;
+            }
+        case BTP_PACS_CHARACTERISTIC_SINK_AUDIO_LOCATIONS:
+            rc = pacs_set_snk_location();
+            if (rc) {
+                return BTP_STATUS_FAILED;
+            }
+        case BTP_PACS_CHARACTERISTIC_SOURCE_AUDIO_LOCATIONS:
+            rc = pacs_set_src_location();
+            if (rc) {
+                return BTP_STATUS_FAILED;
+            }
+        case BTP_PACS_CHARACTERISTIC_AVAILABLE_AUDIO_CONTEXTS:
+            rc = pacs_set_available_contexts();
+            if (rc) {
+                return BTP_STATUS_FAILED;
+            }
+        default:
+            return BTP_STATUS_FAILED;
+    }
+
+    return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t
+pacs_set_supported_contexts(const void *cmd, uint16_t cmd_len,
+                            void *rsp, uint16_t *rsp_len)
+{
+    int rc;
+
+    struct ble_svc_audio_pacs_set_param src_params = {
+        .audio_locations = MYNEWT_VAL(BLE_SVC_AUDIO_PACS_LC3_SRC_AUDIO_LOCATIONS),
+        .supported_contexts = 0
+    };
+    struct ble_svc_audio_pacs_set_param snk_params = {
+        .audio_locations = MYNEWT_VAL(BLE_SVC_AUDIO_PACS_LC3_SNK_SUP_AUDIO_LOCATIONS),
+        .supported_contexts = 0
+    };
+
+    rc = ble_svc_audio_pacs_set(BLE_AUDIO_CODEC_DIR_SOURCE_BIT, &src_params);
+    if (rc) {
+        return BTP_STATUS_FAILED;
+    }
+
+    rc = ble_svc_audio_pacs_set(BLE_AUDIO_CODEC_DIR_SINK_BIT, &snk_params);
+    if (rc) {
+        return BTP_STATUS_FAILED;
+    }
+
+    return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t
+supported_commands(const void *cmd, uint16_t cmd_len,
+                   void *rsp, uint16_t *rsp_len)
+{
+    struct btp_pacs_read_supported_commands_rp *rp = rsp;
+
+    /* octet 0 */
+    tester_set_bit(rp->data, BTP_PACS_READ_SUPPORTED_COMMANDS);
+    tester_set_bit(rp->data, BTP_PACS_UPDATE_CHARACTERISTIC);
+    tester_set_bit(rp->data, BTP_PACS_SET_LOCATION);
+    tester_set_bit(rp->data, BTP_PACS_SET_AVAILABLE_CONTEXTS);
+    tester_set_bit(rp->data, BTP_PACS_SET_SUPPORTED_CONTEXTS);
+
+    *rsp_len = sizeof(*rp) + 2;
+
+    return BTP_STATUS_SUCCESS;
+}
+
+static const struct btp_handler handlers[] = {
+    {
+        .opcode = BTP_PACS_READ_SUPPORTED_COMMANDS,
+        .expect_len = 0,
+        .func = supported_commands,
+    },
+    {
+        .opcode = BTP_PACS_UPDATE_CHARACTERISTIC,
+        .expect_len = BTP_HANDLER_LENGTH_VARIABLE,
+        .func = pacs_update_characteristic,
+    },
+    {
+        .opcode = BTP_PACS_SET_AVAILABLE_CONTEXTS,
+        .expect_len = BTP_HANDLER_LENGTH_VARIABLE,
```

Comment:
those seems to be also fixed size (with no params)

Reply:
Done